Simplecov action fails on forked PRs and dependabot PRs

We need to make sure simplecov action passes for the mentioned categories of PRs.

decoupling simplecov job is not an option as it's throwing an error like this: 
```
Cannot find module '/home/runner/work/supermarket/supermarket/src/supermarket/coverage/.last_run.json'
```

![Screenshot 2021-12-13 at 4.47.54 PM.png](https://images.zenhubusercontent.com/60b7868792ea6cd3231262be/ee75d27b-e0d4-4fb9-aadf-f92389a1c93f)

--- a/.github/workflows/unit.yml
+++ b/.github/workflows/unit.yml
@@ -2,7 +2,7 @@
 name: unit
 
 'on':
-  pull_request:
+  pull_request_target:
   push:
     branches:
       - main

